src/EditorFeatures/CSharpTest/CodeActions/SyncNamespace/SyncNamespaceTests_ChangeNamespace.cs

```diff
@@ -0,0 +1,1858 @@
+@"namespace A.B.C
+{
+    class Class1
+    {
+    }
+
+    class Class2
+    {
+    }
+}";
+            var expectedSourceReference =
+@"
+using A.B.C;
+using Class1Alias = Class1;
+
+namespace Foo
+{
+    using System;
+
+    class RefClass
+    {
+        private Class1Alias c1;
+
+        void M1()
+        {
+            Class2 c2 = null;
+        }
+    }
+}";
+            await TestChangeNamespaceAsync(code, expectedSourceOriginal, expectedSourceReference);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsSyncNamespace)]
+        public async Task ChangeNamespace_WithReferencesInVBDocument()
+        {
+            var defaultNamespace = "A.B.C";
+            var declaredNamespace = "A.B.C.D";
+
+            var documentPath1 = CreateDocumentFilePath(Array.Empty<string>(), "File1.cs");
+            var code =
+$@"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" FilePath=""{ProjectFilePath}"" DefaultNamespace=""{defaultNamespace}"" CommonReferences=""true"">
+        <Document Folders=""{documentPath1.folder}"" FilePath=""{documentPath1.filePath}""> 
+namespace [||]{declaredNamespace}
+{{
+    public class Class1
+    {{ 
+    }}
+}}</Document>
+    </Project>    
+<Project Language=""Visual Basic"" AssemblyName=""Assembly2"" CommonReferences=""true"">
+        <Document> 
+Imports {declaredNamespace}
+
+Public Class VBClass
+    Public ReadOnly Property C1 As Class1
+End Class</Document>
+    </Project>
+</Workspace>";
+
+            var expectedSourceOriginal =
+@"namespace A.B.C
+{
+    public class Class1
+    {
+    }
+}";
+            var expectedSourceReference =
+@"
+Imports A.B.C
+
+Public Class VBClass
+    Public ReadOnly Property C1 As Class1
+End Class";
+            await TestChangeNamespaceAsync(code, expectedSourceOriginal, expectedSourceReference);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsSyncNamespace)]
+        public async Task ChangeNamespace_WithQualifiedReferencesInVBDocument()
+        {
+            var defaultNamespace = "A.B.C";
+            var declaredNamespace = "A.B.C.D";
+
+            var documentPath1 = CreateDocumentFilePath(Array.Empty<string>(), "File1.cs");
+            var code =
+$@"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""Assembly1"" FilePath=""{ProjectFilePath}"" DefaultNamespace=""{defaultNamespace}"" CommonReferences=""true"">
+        <Document Folders=""{documentPath1.folder}"" FilePath=""{documentPath1.filePath}""> 
+namespace [||]{declaredNamespace}
+{{
+    public class Class1
+    {{ 
+    }}
+}}</Document>
+    </Project>    
+<Project Language=""Visual Basic"" AssemblyName=""Assembly2"" CommonReferences=""true"">
+        <Document>
+Public Class VBClass
+    Public ReadOnly Property C1 As A.B.C.D.Class1
+End Class</Document>
+    </Project>
+</Workspace>";
+
+            var expectedSourceOriginal =
+@"namespace A.B.C
+{
+    public class Class1
+    {
```

Comment:
:) nice. this kind of cross language refactoring especially that can involve declaration is always dangerous since change could introduce decleration error and cause skeleton assembly to fail to be created and cause cascading errors. probably fine if you don't do like verification or doesn't care about using staled skeleton assemblies.